src/coreclr/jit/compiler.cpp

```diff
@@ -5157,12 +5157,22 @@ void Compiler::compCompile(void** methodCodePtr, uint32_t* methodCodeSize, JitFl
         m_llvm->Lower();
     });
 
-    fgSsaDomTree = nullptr;
     if (opts.OptimizationEnabled())
     {
+        DoPhase(this, PHASE_DFS_BLOCKS, &Compiler::fgDfsBlocksAndRemove);
+
+        // Clear the immediate dominators so that we process the scratch block.
+        for (BasicBlock* const block : Blocks())
+        {
+            block->bbIDom         = nullptr;
+            block->bbPostorderNum = 0;
+        }
+
+        DoPhase(this, PHASE_COMPUTE_DOMINATORS, &Compiler::fgComputeDominators);
+
         DoPhase(this, PHASE_BUILD_SSA, [this]() {
             lvaComputeRefCounts(/* isRecompute */ true, /* setSlotNumbers */ false);
-            fgResetForSsa();
+
```

Comment:
This deletion doesn't look correct, `fgResetForSsa` clears, e. g. the SSA state of locals.

Reply:
I removed it because of 
```cpp
    // If this is not the first invocation, reset data structures for SSA.
    if (fgSsaPassesCompleted > 0)
    {
        fgResetForSsa();
    }
```
as the first thing in `fgSsaBuild` thinking it was redundant.

Reply:
Great point; so it has probably been redundant all this time.